### PR TITLE
First deliverable

### PR DESCRIPTION
**Pull request by rincon-orlando**
Challenges 1 and 2

### CODE REVIEW

File: coverage_report.txt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    √ validates no filter
+    √ validates videos filter
+    √ validates channels filter (1ms)
+    √ validates unknown filter
+    √ validates exception on invalid data (2ms)
+
+--------------------------|----------|----------|----------|----------|-------------------|
+File                      |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
+--------------------------|----------|----------|----------|----------|-------------------|
+All files                 |    20.45 |       75 |       20 |    20.45 |                   |
```
Comment (PacoMojica) [inline coverage_report.txt:18]: Great 👍  I see the start of a healthy project

File: src/components/Header/Header.styles.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+const Styled = {
+  Container,
+  Search,
+  RightHolder,
+  BurgerButton,
+  ProfileButton,
+  ToggleButton,
+};
+
+export default Styled;
```
Comment (PacoMojica) [inline src/components/Header/Header.styles.js]: Good use of `styled-components`, an alternative to the way you are exporting the components is something like:

```jsx
export const Container = styled.div``;

export const Search = styled.input``;

export const RightHolder = styled.span`
  ....
`;

... rest of components ...
```

this way you can delete from line 50 to 59 because JS will build the object for you, using the `export` before the variable definition will add the value of that variable to the object that will be built and when you need to use a component defined here you can import them like:

```jsx
import { Container, Search, RightHolder, ... } from './Styled';
```

by doing so you won't have to write `Styled.` everytime you want to use a component

File: src/components/VideoCard/VideoCard.component.jsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import Styled from './VideoCard.styles';
+
+const VideoCard = ({ thumbnail, title, body }) => (
+  <Styled.Container>
+    <Styled.Preview src={thumbnail} />
+    <Styled.Title>{title}</Styled.Title>
+    <Styled.Body>{body}</Styled.Body>
+  </Styled.Container>
+);
```
Comment (PacoMojica) [inline src/components/VideoCard/VideoCard.component.jsx]: I like how you defined the `VideoCard`, here you can change the way you export the components from the `VideoCard.styles` and change the import to something like:

```jsx
import { Container, Preview, Title, Body } from './VideoCard.styles';
```

and if you import the components this way you can use them just with the name of the component, like `<Container> ... </Container>`, you don't have to write `Styled.` everytime

File: src/components/VideoCard/VideoCard.styles.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+const Title = styled.h1``;
+
+const Body = styled.p``;
+
+const Preview = styled.img`
+  width: 100%;
+`;
+
+const Styled = { Container, Title, Body, Preview };
+export default Styled;
```
Comment (PacoMojica) [inline src/components/VideoCard/VideoCard.styles.js]: Same comment as before, everything is fine, an improvement you can make is to change the way you export things, it can change to:

```jsx
export const Container = styled.div`
  ...
`;

export const Title = styled.h1``;

export const Body = styled.p``;

export const Preview = styled.img`
  ...
`;
```
and remove the lines 20 & 21, then you can import the components like:

```jsx
import {Container, Title, Body, Preview} from './VideoCard.styles';
```

File: src/utils/tests/videos.spec.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import { getVisibleVideos } from '../videos';
+import mockedData from '../mocked-youtube.json';
+
+const { items } = mockedData;
```
Comment (PacoMojica) [inline src/utils/tests/videos.spec.js:4]: Awesome, you included some edge cases, really good 👍

File: src/utils/videos.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    if (filter === 'channels' && video.id.kind === 'youtube#channel') {
+      return true;
+    }
+
+    if (!filter || filter === 'none') {
+      return true;
+    }
+
+    return false;
+  });
```
Comment (PacoMojica) [inline src/utils/videos.js:16]: Great attention to details 😃

**Review by PacoMojica — COMMENTED** [review]
Hi Orlando 🙂 Your challenge looks great, the header is there, you have tests, the coverage is around 25% you can improve it in future deliveries, good work 👍

**Comment by rincon-orlando** [comment]
Thanks @PacoMojica I fixed your recommendations (except for increasing the coverage, but that will come later) and merged.
